src/server/game/DataStores/DBCStores.cpp

```diff
@@ -844,9 +844,15 @@ CharStartOutfitEntry const* GetCharStartOutfitEntry(uint8 race, uint8 class_, ui
 /// Returns LFGDungeonEntry for a specific map and difficulty. Will return first found entry if multiple dungeons use the same map (such as Scarlet Monastery)
 LFGDungeonEntry const* GetLFGDungeon(uint32 mapId, Difficulty difficulty)
 {
-    for (LFGDungeonEntry const* dungeon : sLFGDungeonStore)
-        if (dungeon->map == int32(mapId) && Difficulty(dungeon->difficulty) == difficulty)
+    for (uint32 i = 0; i < sLFGDungeonStore.GetNumRows(); ++i)
+    {
+        LFGDungeonEntry const* dungeon = sLFGDungeonStore.LookupEntry(i);
+        if (!dungeon)
+            continue;
+
+        if (dungeon->MapID == int32(mapId) && Difficulty(dungeon->Difficulty) == difficulty)
```

Comment:
why do you need to convert it to int32? uin32 is just fine?

Reply:
Do we have a case where the interger will be negative?

Reply:
unless you are hinting i should change
```
LFGDungeonEntry const* GetLFGDungeon(uint32 mapId, Difficulty difficulty)
```
to
```
LFGDungeonEntry const* GetLFGDungeon(int32 mapId, Difficulty difficulty)
``` 
and respected spots? but that if was already like that originally.